### PR TITLE
Fix the way we target linting at specific path

### LINKED ISSUE #4117: Github build validation fails on missing tags when it shouldn't
**Describe the bug**
We enforce tags for Azure resources at different levels of the solution and our build validation should take that under advisement but it doesn't... Is requires `tre_workspace_id` tag on a `devops` level resource which isn't reasonable.

### PR DESCRIPTION
**Pull request by tamirkamara**
# Resolves #4117 

## What is being addressed

Linting tags was supposed to target specific paths for each pass, but apparently it didn't or a update caused it to malfunction.

## How is this addressed

- Update the environment variable name used to send the path to the linter

### CODE REVIEW

**Review by marrobi — APPROVED** [review]
LGTM if it works. 😊

**Comment by tamirkamara** [comment]
> LGTM if it works. 😊

It does :-)

**Review by yuvalyaron — APPROVED** [review]
🚀

File: .github/workflows/build_validation_develop.yml [hunk 1]
```diff
@@ -104,7 +118,7 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           VALIDATE_TERRAFORM_TFLINT: true
           TERRAFORM_TFLINT_CONFIG_FILE: .tflint_workspaces.hcl
-          LINTER_REGEX_INCLUDE: './templates/workspaces/.*'
+          FILTER_REGEX_INCLUDE: './templates/workspaces/.*'
```
[the remark this answers is not in the document]
Comment (tamirkamara) [reply .github/workflows/build_validation_develop.yml:121]: The short answer is that `LINTER_REGEX_INCLUDE` might have not ever worked. Or, at the very least it's name has changed at some point.
`FILTER_REGEX_INCLUDE` is to limit the files the linter goes over but when run in an _action_ it applies on top of files that were changed. 
There's a condition at the top of each step that only runs it if there were Terraform changes in the folder. This is to save time as the action uses a large docker image and we don't want to load it if the linter anyway won't find changed files per the `FILTER_REGEX_INCLUDE`

**Review by tim-allen-ck — APPROVED** [review]
LGTM

Happy to merge when you are @tamirkamara

**Pull request merged**